pkg/api/types.go

```diff
@@ -388,8 +391,24 @@ const (
 
 	// MountOptionAnnotation defines mount option annotation used in PVs
 	MountOptionAnnotation = "volume.beta.kubernetes.io/mount-options"
+
+	// StorageTopologyConstraint defines topology constraints of a PV
+	// Value is type []TopologyConstraint
```

Comment:
I'm thinking of replacing this with the NodeAffinity type instead, since I think this can be expressed through NodeSelector.  Then we don't have to add this new type for topology constraints.  For alpha at least and maybe even beta, we will only implement the RequiredDuringSchedulingIgnoredDuringExecution NodeSelector part and ignore/error on the other fields.

What do you guys think?